feat(KaizenProvider): add KaizenProvider to @kaizen/components (#3239)
cultureamp/kaizen-design-system: packages/components/src/KaizenProvider/docs/Managing-the-theme.stories.mdx

import { Meta } from '@storybook/addon-docs'
import { CATEGORIES } from "../../../../../storybook/constants"

<Meta title={`${CATEGORIES.components}/KaizenProvider/Managing the theme`} />

# Managing the theme

Out of the box, the `KaizenProvider` utilises the `defaultTheme` from `@kaizen/design-tokens`. However if you want to roll your own theme, we provide a `ThemeManager` function which can apply your custom theme.

```jsx
import { ThemeManager } from '@kaizen/components'

export const themeManager = new ThemeManager({
  themeKey: "custom",
  spacing: {
    md: "1rem",
  },
});

<KaizenProvider themeManager={themeManager}>
  <App />
</KaizenProvider>
```

## How to switch themes

> You can use the `setAndApplyTheme` method to change the current theme, to either Heart or a customised object.

Reviewer: ```suggestion
You can use the `setAndApplyTheme` method to change the current theme to either Heart or a customised object.
```

Example:

```js
themeManager.setAndApplyTheme(heartTheme)

// or make your own modifications
themeManager.setAndApplyTheme({
  ...heartTheme,
  themeKey: "custom",
  spacing: {
    ...heartTheme.spacing,
    md: "1rem",
  },
})

```

## Using the theme in apps

> For more details on how theme-ing works, checkout our [design tokens documentation](/story/design-tokens-how-to-use-design-tokens-in-react-components--page).

Reviewer: There's one here 😛 Something like this?
```suggestion
For more details on how to apply a theme, checkout our [design tokens documentation](/story/design-tokens-how-to-use-design-tokens-in-react-components--page).
```
